src/math.F90

```diff
@@ -6,127 +6,144 @@ module math
+      implicit none
+      integer(C_INT), value, intent(in) :: n
+      real(C_DOUBLE), intent(in)  :: uvw(3)
+      real(C_DOUBLE), intent(out) :: rn(2 * n + 1)
+    end subroutine calc_rn_c_intfc
 
-  end function normal_percentile
+    pure subroutine calc_zn_c_intfc(n, rho, phi, zn) bind(C, name='calc_zn_c')
+      use ISO_C_BINDING
+      implicit none
+      integer(C_INT), value, intent(in) :: n
+      real(C_DOUBLE), value, intent(in) :: rho
+      real(C_DOUBLE), value, intent(in) :: phi
+      real(C_DOUBLE), intent(out) :: zn(((n + 1) * (n + 2)) / 2)
+    end subroutine calc_zn_c_intfc
 
-!===============================================================================
-! T_PERCENTILE calculates the percentile of the Student's t distribution with a
-! specified probability level and number of degrees of freedom
-!===============================================================================
+    subroutine rotate_angle_c_intfc(uvw, mu, phi) bind(C, name='rotate_angle_c')
+      use ISO_C_BINDING
+      implicit none
+      real(C_DOUBLE), intent(inout) :: uvw(3)
+      real(C_DOUBLE), value, intent(in)    :: mu
+      real(C_DOUBLE), optional, intent(in) :: phi
+    end subroutine rotate_angle_c_intfc
 
-  elemental function t_percentile(p, df) result(t)
+    function maxwell_spectrum_c_intfc(T) bind(C, name='maxwell_spectrum_c') &
+         result(E_out)
+      use ISO_C_BINDING
+      implicit none
+      real(C_DOUBLE), value, intent(in) :: T
+      real(C_DOUBLE) :: E_out
+    end function maxwell_spectrum_c_intfc
 
-    real(8), intent(in) :: p  ! probability level
-    integer, intent(in) :: df ! degrees of freedom
-    real(8)             :: t  ! corresponding t-value
+    function watt_spectrum_c_intfc(a, b) bind(C, name='watt_spectrum_c') &
+         result(E_out)
+      use ISO_C_BINDING
+      implicit none
+      real(C_DOUBLE), value, intent(in) :: a
+      real(C_DOUBLE), value, intent(in) :: b
+      real(C_DOUBLE) :: E_out
+    end function watt_spectrum_c_intfc
 
-    real(8)            :: n  ! degrees of freedom as a real(8)
-    real(8)            :: k  ! n - 2
-    real(8)            :: z  ! percentile of normal distribution
-    real(8)            :: z2 ! z * z
+    function faddeeva_c_intfc(z) bind(C, name='faddeeva_c') result(wv)
+      use ISO_C_BINDING
+      implicit none
+      complex(C_DOUBLE_COMPLEX), value, intent(in) :: z
+      complex(C_DOUBLE_COMPLEX) :: wv
+    end function faddeeva_c_intfc
 
-    if (df == 1) then
-      ! For one degree of freedom, the t-distribution becomes a Cauchy
-      ! distribution whose cdf we can invert directly
+    function w_derivative_c_intfc(z, order) bind(C, name='w_derivative_c') &
+           result(wv)
+      use ISO_C_BINDING
+      implicit none
+      complex(C_DOUBLE_COMPLEX), value, intent(in) :: z
+      integer(C_INT), value,            intent(in) :: order
+      complex(C_DOUBLE_COMPLEX) :: wv
+    end function w_derivative_c_intfc
```

Comment:
Looks like `faddeeva_c` and `w_derivative_c` don't exist.

Reply:
- [x]